src/main/java/scmspain/karyon/restrouter/RestRouterHandler.java

```diff
@@ -0,0 +1,222 @@
+
+    Route<ByteBuf, ByteBuf> route =  restUriRouter.findBestMatch(request, response)
+        .orElse(new RouteNotFound<>());
+    Observable<Void> result;
+
+    if (route.isCustomSerialization()) {
+      result = handleCustomSerialization(route, request, response);
+
+    } else {
+      result = handleSupported(route, request, response);
+
+    }
+
+    return result;
+  }
+
+
+  private Observable<Void> handleSupported(Route<ByteBuf,ByteBuf> route,
+                                          HttpServerRequest<ByteBuf> request,
+                                          HttpServerResponse<ByteBuf> response) {
+    Observable<Object> resultObs;
+    Optional<String> negotiatedContentType;
+    Set<String> supportedContentTypes = getSupportedContentTypes(route);
+
+    try {
+      negotiatedContentType = Optional.of(acceptNegotiation(request, supportedContentTypes));
+      resultObs = route.getHandler().process(request, response);
+
+    } catch (CannotSerializeException | InvalidAcceptHeaderException e) {
+      resultObs = Observable.error(e);
+      negotiatedContentType = Optional.empty();
+    }
+
+    String contentType = negotiatedContentType
+        .orElseGet(serializerManager::getDefaultContentType);
+
+    resultObs = resultObs.onErrorResumeNext(throwable -> {
+          if (errorHandler != null) {
+            return errorHandler.handleError(request, throwable, response::setStatus);
+
+          } else {
+            return Observable.error(throwable);
+          }
+        }
+    );
+
+    // If RouteNotFound is not handle it will be handled here
+    resultObs = resultObs.onErrorResumeNext(throwable -> {
+      return defaultErrorHandler.handleError(request, throwable, response::setStatus);
+    });
+
+    Serializer serializer = serializerManager.getSerializer(contentType)
+        .orElseThrow(() -> new CannotSerializeException("Cannot serialize " + contentType));
+
+    SerializeWriter writer = new SerializeWriter(response, contentType);
+
+    return resultObs.flatMap(result -> writer.write(result, serializer));
+  }
+
+
+  private Observable<Void> handleCustomSerialization(Route<ByteBuf, ByteBuf> route,
+                                                    HttpServerRequest<ByteBuf> request,
+                                                    HttpServerResponse<ByteBuf> response) {
+
+    Observable<Object> resultObs = route.getHandler()
+        .process(request, response);
+
+    resultObs = resultObs.onErrorResumeNext(throwable -> {
+      return defaultErrorHandler.handleError(request, throwable, response::setStatus)
+          .map(this::serializeErrorDto)
+          .flatMap(response::writeStringAndFlush);
+    });
+
+    return resultObs.cast(Void.class);
+  }
+
+  private String serializeErrorDto(RestRouterErrorDTO errorDTO) {
+    return "{\"description\":\"" + errorDTO.getDescription() + "\",\"timestamp\":\"" + errorDTO.getTimestamp() + "\"}";
```

Comment:
Because this error handling last resource is used even if you don't use the serialization configuration.